invenio_pidstore/provider.py

```diff
@@ -48,6 +48,14 @@ class PIDStatus:
     registered.
     """
 
+    @classmethod
+    def list_in_order(cls):
```

Comment:
I agree that Admin UI will benefit from translation on this one, so we should do it. I'm just concerned wondering where is the best place to put, perhaps just directly in admin model view:

```Python
# invenio_pidstore/admin.py
class PersistentIdentifierModelView(ModelView):
    form_choices = {
        'status': ((PIDStatus.NEW, _('New')),
                      (PIDStatus.RESERVED, _('Reserved')), ...)
    }
```
I'm not even sure whether the ordered listing of of PIDStatus is going to be reusable anyway.